module/trie.c

```diff
@@ -269,14 +306,528 @@ int trie_count_completion(struct trie *t, char *pre)
     return trie_count_completion_recursive(end);
 }
 
+/* 
+    Since modules do not include header files typically, this is an early 
+    declaration of the suggestions() function since helper functions use it
+ */
+int suggestions(match_t **set, struct trie *t, char *prefix, char *suffix, int edits_left, int n);
+
+/*
+    Sees if there are any words in a trie that contain a given prefix
+  
+    Parameters: 
+     - t: A trie. Must point to a trie allocated with trie_new
+     - s: A string. The prefix to check
+  
+    Returns:
+     - true if there is a word in the dictionary with the prefix
+     - false otherwise
+*/
+bool has_children(struct trie *t, char *s) 
+{    
+    if (trie_search(t, s) == NOT_IN_TRIE) {
```

Comment:
Why not just `return trie_search(t, s) == NOT_IN_TRIE`?

Reply:
That is a lot more elegant, good catch.